### PR TITLE
perf(rating): remove ngClass, ngRepeat, event directives and isolated…

### LINKED ISSUE #16: perf(rating): do not rely on ngRepeat

### PR DESCRIPTION
**Pull request by mxth**
… scope

Close #16, #17

@m0t0r, @caitp it looks lighter than before :)

### CODE REVIEW

File: src/elements/rating/rating.js [hunk 1]
```diff
@@ -13,75 +13,133 @@
...
+    }
   }
 
-  smRatingController.$inject = ['$scope', '$attrs', 'ratingConfig'];
+  smRatingController.$inject = ['$scope', '$element', '$attrs', 'ratingConfig', '$parse'];
+
+  function smRatingController($scope, $element, $attrs, ratingConfig, $parse) {
+    var self = this;
+
+    self.$scope = $scope;
```
Comment (m0t0r) [inline src/elements/rating/rating.js]: @mxth any particular reason to attach everything to _this_ ?

File: src/elements/rating/rating.js [hunk 2]
```diff
@@ -13,75 +13,133 @@
...
+    self.$element = $element;
+    self.$attrs = $attrs;
+    self.ratingConfig = ratingConfig;
+
+    self.ngModelCtrl  = { $setViewValue: angular.noop };
```
Comment (m0t0r) [inline src/elements/rating/rating.js]: I am not sure why we need to do this. Can you explain a bit ?

File: src/elements/rating/rating.js (hunk 1 above)
Comment (caitp) [inline src/elements/rating/rating.js]: Not in favor of this line

Comment (mxth) [inline src/elements/rating/rating.js]: Not everything, just the ones that need accessing from prototype functions

File: src/elements/rating/rating.js (hunk 2 above)
Comment (mxth) [inline src/elements/rating/rating.js]: I borrowed this from ui-bootstrap. I think the purpose is to let `rate` function be called before `link` without throwing `undefined` error

**Pull request closed without merging**